src/transformations.jl

```diff
@@ -8,63 +8,106 @@ abstract type AbstractDataTransform end
 """
-transform!(t::AbstractDataTransform, x::AbstractArray{<:Real,1}) = transform!(x, t, x)
-transform!(t::AbstractDataTransform, x::AbstractArray{<:Real,2}) = transform!(x, t, x)
+function transform!(t::AbstractDataTransform, x::AbstractMatrix{<:Real})
+    if t.dims == 1
+        transform!(x, t, x)
+    elseif t.dims == 2
+        transform!(x', t, x')
+    end
+    return x
+end
+
+function transform!(t::AbstractDataTransform, x::AbstractVector{<:Real})
+    transform!(t, reshape(x, :, 1))
+    return vec(x)
+end
 
 """
     transform(t::AbstractDataTransform, x)
 
-Return a row-standardized vector or matrix `x` using `t` transformation.
+Return a standardized vector or matrix `x` using `t` transformation.
 """
-transform(t::AbstractDataTransform, x::AbstractArray{<:Real,1}) = transform!(similar(x), t, x)
-transform(t::AbstractDataTransform, x::AbstractArray{<:Real,2}) = transform!(similar(x), t, x)
+function transform(t::AbstractDataTransform, x::AbstractMatrix{<:Real})
+    if t.dims == 1
+        return transform!(similar(x), t, x)
+    elseif t.dims == 2
+        return transform!(similar(x)', t, x')'
+    end
+end
+
+function transform(t::AbstractDataTransform, x::AbstractVector{<:Real})
+    xmat = transform(t, reshape(x, :, 1))
+    return vec(xmat)
+end
 
 # reconstruct the original data from transformed values
 """
     reconstruct!(t::AbstractDataTransform, y)
 
-Perform an in-place reconstruction into an original data scale from a row-transformed
+Perform an in-place reconstruction into an original data scale from a transformed
 vector or matrix `y` using `t` transformation.
 """
-reconstruct!(t::AbstractDataTransform, y::AbstractArray{<:Real,1}) = reconstruct!(y, t, y)
-reconstruct!(t::AbstractDataTransform, y::AbstractArray{<:Real,2}) = reconstruct!(y, t, y)
+function reconstruct!(t::AbstractDataTransform, y::AbstractMatrix{<:Real})
+    if t.dims == 1
+        reconstruct!(y, t, y)
+    elseif t.dims == 2
+        reconstruct!(y', t, y')
+    end
+    return y
+end
+
+function reconstruct!(t::AbstractDataTransform, y::AbstractVector{<:Real})
+    reconstruct!(t, reshape(y, :, 1))
+    return vec(y)
+end
 
 """
     reconstruct(t::AbstractDataTransform, y)
 
-Return a reconstruction of an originally scaled data from a row-transformed vector
+Return a reconstruction of an originally scaled data from a transformed vector
 or matrix `y` using `t` transformation.
 """
-reconstruct(t::AbstractDataTransform, y::AbstractArray{<:Real,1})  = reconstruct!(similar(y), t, y)
-reconstruct(t::AbstractDataTransform, y::AbstractArray{<:Real,2})  = reconstruct!(similar(y), t, y)
+function reconstruct(t::AbstractDataTransform, y::AbstractMatrix{<:Real})
+    if t.dims == 1
+        return reconstruct!(similar(y), t, y)
+    elseif t.dims == 2
+        return reconstruct!(similar(y)', t, y')'
+    end
+end
+
+function reconstruct(t::AbstractDataTransform, y::AbstractVector{<:Real})
+    ymat = reconstruct(t, reshape(y, :, 1))
```

Comment:
```suggestion
    vec(reconstruct(t, reshape(y, :, 1)))
```
and remove next line.